contracts/PartyBid.sol

```diff
@@ -0,0 +1,701 @@
+        // validate that a portion of the token supply is not being burned
+        if (_partyHost == address(0)) {
+            require(_partyHostBasisPoints == 0, "PartyBid::initialize: can't send tokens to burn addr");
+        }
+        partyHostBasisPoints = _partyHostBasisPoints;
+        partyHost = _partyHost;
+        // validate token exists
+        require(_getOwner() != address(0), "PartyBid::initialize: NFT getOwner failed");
+        // validate auction exists
+        require(
+            IMarketWrapper(_marketWrapper).auctionIdMatchesToken(
+                _auctionId,
+                _nftContract,
+                _tokenId
+            ),
+            "PartyBid::initialize: auctionId doesn't match token"
+        );
+    }
+
+    // ======== External: Contribute =========
+
+    /**
+     * @notice Contribute to the PartyBid's treasury
+     * while the auction is still open
+     * @dev Emits a Contributed event upon success; callable by anyone
+     */
+    function contribute() external payable nonReentrant {
+        require(
+            partyStatus == PartyStatus.AUCTION_ACTIVE,
+            "PartyBid::contribute: auction not active"
+        );
+        address _contributor = msg.sender;
+        uint256 _amount = msg.value;
+        require(_amount > 0, "PartyBid::contribute: must contribute more than 0");
+        // get the current contract balance
+        uint256 _previousTotalContributedToParty = totalContributedToParty;
+        // add contribution to contributor's array of contributions
+        Contribution memory _contribution =
+            Contribution({
+                amount: _amount,
+                previousTotalContributedToParty: _previousTotalContributedToParty
+            });
+        contributions[_contributor].push(_contribution);
+        // add to contributor's total contribution
+        totalContributed[_contributor] =
+            totalContributed[_contributor] +
+            _amount;
+        // add to party's total contribution & emit event
+        totalContributedToParty = totalContributedToParty + _amount;
+        emit Contributed(
+            _contributor,
+            _amount,
+            _previousTotalContributedToParty,
+            totalContributed[_contributor]
+        );
+    }
+
+    // ======== External: Bid =========
+
+    /**
+     * @notice Submit a bid to the Market
+     * @dev Reverts if insufficient funds to place the bid and pay PartyDAO fees,
+     * or if any external auction checks fail (including if PartyBid is current high bidder)
+     * Emits a Bid event upon success.
+     * Callable by any contributor
+     */
+    function bid() external nonReentrant {
+        require(
+            partyStatus == PartyStatus.AUCTION_ACTIVE,
+            "PartyBid::bid: auction not active"
+        );
+        require(
+            totalContributed[msg.sender] > 0,
+            "PartyBid::bid: only contributors can bid"
+        );
```

Comment:
yeah we've talked about this a good deal in various forums, it's not really necessary, it puts a small barrier to entry of someone having to pay gas fees to contribute 1 wei to unlock this power if they are not an actual contributor, but on the flipside you could argue it would be better to be permissionless so that a bot could be built to do this on behalf of users, or on the other end of the spectrum you could argue that someone should have some % of the total contributions in order to be able to bid.
all of the conversations have pretty much netted out with, "you could argue that one approach or another is better, we are going to leave it as-is for now"